### ISSUE TITLE
weight clip bad behavior

### ISSUE THREAD

**Issue by nchiprut** [larq/larq#544]
### Describe the bug
adding weight clip on a simple kernel-quantized network does not train properly with larq version 0.10.0 and tf version 2.3.0,
when reverting back to tf 2.2.0 and larq 0.9.6, train works as expected.
by "not properly" I mean - stays at random accuracy, and by "as expected" I mean 0.86 accuracy on MNIST with the code below,

important to notice that by commenting weight_constraint with larq version 0.10.0 and tf version 2.3.0, the net also trains as expected.

### To Reproduce
```
quant_model = tf.keras.models.Sequential([
    tf.keras.layers.InputLayer(input_shape=input_shape),
    tf.keras.layers.Flatten(),
    lq.layers.QuantDense(num_classes,
              kernel_quantizer='ste_sign',
              kernel_constraint="weight_clip",
              use_bias=False),
    tf.keras.layers.Activation("softmax")

])

quant_model.compile(
    optimizer='adam',
    loss="categorical_crossentropy",
    metrics=["accuracy"],
)
trained_model = quant_model.fit(
        train_batches,
        epochs=n_epochs,
        validation_data=test_batches,
    )

```
### Expected behavior
to work as previous version.

### Environment

TensorFlow version: 2.3.0
Larq version: 0.10.0

**Comment by lgeiger** [comment]
@nchiprut Thanks for opening the issue.
Would you be able to provide the full code example? I can take a look at this problem tomorrow then.

**Comment by lgeiger** [comment]
@nchiprut are you still running into this issue? Would you be able to provide the complete code example, so we can reproduce the problem?

**Comment by leonoverweel** [comment]
Closing due to inactivity for now; @nchiprut feel free to reopen if you're still having this problem and would like us to take a look at the code.

**Comment by prapti1998** [comment]
@nchiprut were you able to solve this issue?
